ast/util.go

```diff
@@ -62,14 +66,102 @@ func (checker *readOnlyChecker) Leave(in Node) (out Node, ok bool) {
 	return in, checker.readOnly
 }
 
-// WriteName append escaped `name` with back quote to `sb`.
-func WriteName(sb *strings.Builder, name string) {
-	sb.WriteString("`")
-	sb.WriteString(EscapeName(name))
-	sb.WriteString("`")
+//RestoreFlag mark the Restore format
+type RestoreFlags uint64
+
+const (
+	RestoreStringSingleQuotes RestoreFlags = 1 << iota
+	RestoreStringDoubleQuotes
+	RestoreStringEscapeBackslash
+
+	RestoreKeyWordUppercase
+	RestoreKeyWordLowercase
+
+	RestoreNameUppercase
+	RestoreNameLowercase
+	RestoreNameOriginal
+	RestoreNameDoubleQuotes
+	RestoreNameBackQuote
+)
+
+const (
+	DefaultRestoreFlags = RestoreStringSingleQuotes | RestoreKeyWordUppercase | RestoreNameOriginal |
+		RestoreNameBackQuote
+)
+
+// Has return weather `rf` has this flag
+func (rf RestoreFlags) Has(flag RestoreFlags) bool {
```

Comment:
I think we'd better split this function into small ones.
e.g. parser/mysql/types.go HasXXXFlag() bool